runtime/lua/vim/shared.lua

```diff
@@ -986,4 +986,56 @@ do
   end
 end
 
+---@param tabpage integer
+---@param layout table
+function vim._set_layout(tabpage, layout)
+  ---@param node { [1]: string, [2]: string|integer }
+  ---@return integer?
+  local function set_layout(node)
+    local ty = node[1]
+    if ty == 'leaf' then
+      local buf = node[2]
+      if type(buf) == 'string' then
+        buf = vim.fn.bufadd(buf)
+        vim.api.nvim_set_option_value('buflisted', true, {
+          buf = buf,
+        })
+      end
+      local win = vim.api.nvim_tabpage_get_win(tabpage)
+      vim.api.nvim_win_set_buf(win, buf)
+      -- If node[3] is truthy, the window is focused.
+      if node[3] then
```

Comment:
This could later be changed to a dictionary to add additional properties to windows in the layout (for setting `topline`, etc.). Should this be done in another PR, or in this one?

This could be changed to take a dict like `{ focused = true }` instead of just a boolean, which would allow for updating this function without breaking configs later on.